lib/tramway/navbar.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Tramway
+  # Navbar class for Tramway navbars
+  #
+  class Navbar
+    attr_reader :items
+
+    def initialize(context)
+      @items = {}
+      @context = context
+    end
+
+    def left
+      return unless block_given?
+
+      @items[:left] = []
+
+      @filling = :left
+
+      yield self
+    end
+
+    def right
+      return unless block_given?
+
+      @items[:right] = []
+
+      @filling = :right
```

Comment:
a bit strange implementation with this flag. Is it a common way to do this?

Reply:
I don't know about a common way. `@filling` is like a `filling_mode`.

```ruby
@filling = :right # means that NOW we fill the right part
```

Reply:
I understand. I mean the common way for doing these things with nested scopes

```
  nav.left do
    nav.item 'Users', '/users'
    nav.item 'Podcasts', '/podcasts'
  end
```

Usage of instance variable for that feels a bit wrong(though I don't see atm if it could cause any troubles)

Reply:
There is just one workaround here. It's decomposing code block 

```ruby
do
    nav.item 'Users', '/users'
    nav.item 'Podcasts', '/podcasts'
end
```

to lines and process each one. I think this way is not very good, cause even Ruby does not provide an access to each line of code block by default.